### ISSUE TITLE
compact-editor addon: customize editor tabs (text only, icons only)

### ISSUE THREAD

**Issue by WorldLanguages** [ScratchAddons/ScratchAddons#6432]
### Why this would be helpful

PR #5866 (closed) proposed a feature where only the icons are visible within the tabs:
![image](https://github.com/ScratchAddons/ScratchAddons/assets/17484114/75bc50de-c114-453e-954e-a2ff9bafe631)

New addon "compact editor" (#4018) removes the icons from the tabs without a setting to change this behavior:
![image](https://github.com/ScratchAddons/ScratchAddons/assets/17484114/fd075c81-d18b-4eec-8e71-954fdea18135)

### My suggestion

Provide a new setting for the compact-editor addon:

1. Default: text only
2. Icons only (we should discuss how much padding to add to the sides of the icons so that the buttons are still easily clickable)
3. Possibly, an option to keep both the icons and text with compact-editor enabled.

**Comment by CST1229** [comment]
Reposting: https://github.com/ScratchAddons/ScratchAddons/pull/4018#issuecomment-1646887286
> I don't really see any use for that (since there isn't any overflow on the tab bar even with find bar and variable manager enabled on the lowest possible editor width).

Also, Scratch 2.0 and 1.4 didn't have icon-only tabs (in fact, they had *text*-only tabs).

**Comment by WorldLanguages** [comment]
We could also consider a setting that keeps the icons in the editor menu bar menus:

![image](https://github.com/ScratchAddons/ScratchAddons/assets/17484114/b0fc8455-b02d-402b-b3da-d06d70df8f6b)

**Comment by Samq64** [comment]
Related: #6297

**Comment by WorldLanguages** [comment]
> Related: #6297

Good point! The editor menu bar customizations probably belong in a "custom editor menu bar" addon.

**Comment by WorldLanguages** [comment]
> New addon "compact editor" (#4018) removes the icons from the tabs without a setting to change this behavior:

How should we handle this? We already provide an addon that lets you hide the tutorials button, and we'll probably extend it to do more things (see PR #6297) such as:
1. Add a setting to hide the My Stuff button
2. Support the "compact user dropdown" functionality from discuss-button inside the editor (different toggle though)
3. **Add a setting to replace the settings/file/editor menus with just the icons, *or potentially just the text***

Will users be interested on keeping just the text without enabling other style changes from "compact editor"? If both addons can change the editor menu bar, which one should have priority?
 
Personally, I don't think we should remove the editor menu bar changes from compact-editor. But that leaves the "customizable editor menu bar" addon in a quite awkward position.

**Comment by iqnite** [comment]
> Personally, I don't think we should remove the editor menu bar changes from compact-editor. But that leaves the "customizable editor menu bar" addon in a quite awkward position.

Not *remove*, just *move*. And we should do that before the stable version gets released, so users don't get confused.
Take LibreOffice as an example. It has a Button setting with 3 options: Show Label and Icon, Icon only, and Label only.
That's why I used a dropdown setting instead of a boolean. I wanted to add a way to hide icons but didn't know how to get them via CSS. Now that @lisa-wolfgang has made it, I could add the option. (but I still don't know which part of the userstyle does that)

Generally: Scratch Addons' purpose is to be *modular*. The more settings there are, the better it is. Let's not repeat the mistake we did when we put navbar customization options in the 2.0 -> 3.0 addon.

**Comment by WorldLanguages** [comment]
> Not _remove_, just _move_.

Yeah, I think we're talking about the same concept, call it however you like.

> The more settings there are, the better it is. 

Yeah, I don't think anyone argued against this setting, I just wanted to discuss where it would belong. Maybe we should have changed "compact editor" CSS before the v1.34.0 release, but it's too late now.

**Comment by iqnite** [comment]
> it's too late now.

Why is it too late?

**Comment by Samq64** [comment]
> > it's too late now.
> 
> Why is it too late?

It was already released to the stores.

**Comment by iqnite** [comment]
It can still be migrated, just like the "Hide Tutorials button" addon.
